### DISCUSSION TITLE
Storage

### DISCUSSION THREAD

**Discussion by Mikaleb** [Mikaleb/Dashlife#3] [General]
Per [this article from Google](https://web.dev/storage-for-the-web/)  from Pete LePage, the best solution so far for storing the data of Dashlife locally is the IndexedDB browser functionality.

> What about user wanting cross-device ? 

To anticipate a bit, the day users need to have their data cross browser, we need to make a module to export theses datas in a simple format (json surely) to be passed onto any other device. The security layer to prevent the file to be stolen shall be to generate a sentence to copy (or user generated) to encrypt the file, then when the user import say file, shall be asked the key to be able to import.

TLDR; generate a key, hash, export to json. Then ask for the key, decrypt, import json datas.

Of course if users want to avoid this hassle to cross-sync, we can always setup a storage solution via a paid solution but it'll depend on the demand.
